### PR TITLE
Refactor options screen

### PR DESCRIPTION
**Pull request by Flamefire**
Extracted from #1553

I needed some more space and a way to ease adding a new option.

New:

### CODE REVIEW

**Comment by Spikeone** [comment]
It'd be great to have an interface option (left of zurück) where some interface options are (like showgameframe info) :)

**Comment by Flow86** [comment]
perhaps we should even split out a "Network" category? and put the network stuff there?

**Comment by Flamefire** [comment]
Makes sense. Maybe open an issue for that so someone with the time for it can do that :) Would be an easy task for first-time contributors I guess.

File: libs/s25main/desktops/dskOptions.cpp [hunk 1]
```diff
@@ -222,76 +236,65 @@ dskOptions::dskOptions() : Desktop(LOADER.GetImageN("setup013", 0))
...
-                               NormalFont);
+    optiongroup->AddTextButton(ID_btOn, curPos + ctrlOffset, ctrlSize, TextureColor::Grey, _("On"), NormalFont);
+    optiongroup->AddTextButton(ID_btOff, curPos + ctrlOffset2, ctrlSize, TextureColor::Grey, _("Off"), NormalFont);
+    curPos.y += 50;
 
     // "Grafiktreiber"
-    groupGrafik->AddText(ID_txtVideoDriver, DrawPoint(80, 275), _("Graphics Driver"), COLOR_YELLOW, FontStyle{},
-                         NormalFont);
-    combo = groupGrafik->AddComboBox(ID_cbVideoDriver, DrawPoint(280, 275), Extent(390, 20), TextureColor::Grey,
+    groupGrafik->AddText(ID_txtVideoDriver, curPos, _("Graphics Driver"), COLOR_YELLOW, FontStyle{}, NormalFont);
```
Comment (falbrechtskirchinger) [inline libs/s25main/desktops/dskOptions.cpp:296]: Incidentally fixes the off-by-5-pixels error of the "Graphics Driver" label. :+1:

File: tests/s25Main/integration/testDskGameInterface.cpp
Comment (falbrechtskirchinger) [inline tests/s25Main/integration/testDskGameInterface.cpp:1]: I don't think you meant to include this file here when you extracted this PR from #1553, right?

Comment (falbrechtskirchinger) [reply tests/s25Main/integration/testDskGameInterface.cpp:1]: I was confused why the build was succeeding with that test. I see that `revert_mouse` is already a valid setting, it's just not configurable in the options screen. :bulb:

Comment (Flamefire) [reply tests/s25Main/integration/testDskGameInterface.cpp:1]: Indeed. Thanks!

**Comment by Flamefire** [comment]
@Flow86 Can this be merged? Required for #1553

**Review by Flow86 — APPROVED** [review]

**Comment by Flamefire** [comment]
@Flow86 The required check "Linux (clang, macos-12, Debug)" needs to be changed to "Linux (clang, macos-12, Debug, 1.82.0)"

And what is the issue on Jenkins? Can't login right now to check.

**Comment by falbrechtskirchinger** [comment]
#1594 and #1602 are also waiting on this.

**Comment by Flow86** [comment]
Oh I thought this one was already merged.

**Pull request merged**